### PR TITLE
Create security_test.yml and move default credentials for api in it

### LINKED ISSUE #31694: The new API is available for everyone with default credential & default API keys
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Actually, it's possible to use the new API (there is no Feature Flag for this experimental feature) in developer mode with default credentials (https://github.com/PrestaShop/PrestaShop/blob/8.1.x/app/config/security_dev.yml#L16).

### Expected behavior

_No response_

### Steps to reproduce

1. Enable developer mode
2. Target the http://domain.tld/admin-dev/api/oauth2/token with credentials 
```
{
          client_id: 'my_client_id',
          client_secret: 'prestashop',
          grant_type: 'client_credentials',
}
```
3. You obtain a valid access token.

### PrestaShop version(s) where the bug happened

8.1.x

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by mflasquin**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Create security_test.yml and move default credentials for api in it
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Old credentials not works anymore. Add them in security_dev.yml to make new api works
| Fixed ticket?     | Fixes #31694 
| Related PRs       | -
| Sponsor company   | Prestashop SA

### CODE REVIEW

File: app/config/security_test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# http://symfony.com/doc/current/book/security.html
+security:
+  # http://symfony.com/doc/current/book/security.html#where-do-users-come-from-user-providers
+  providers:
+    in_memory:
+      memory: ~
+    admin:
+      id: prestashop.security.admin.provider
+
+    # Temporary provider & encoder to be able to use the new BO API in dev mode
```
Comment (jolelievre) [inline app/config/security_test.yml]:
```suggestion
    # Temporary provider & encoder to be able to use the new BO API in test mode
```

**Comment by mflasquin** [comment]
ga.tests.ui.pr : https://github.com/mflasquin/ga.tests.ui.pr/actions/runs/4489147955

**Review by jolelievre — DISMISSED** [review]
Thanks @mflasquin

File: app/config/security_test.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# To get started with security, check out the documentation:
+# http://symfony.com/doc/current/book/security.html
```
Comment (matks) [inline app/config/security_test.yml]:
```suggestion
# https://symfony.com/doc/current/security.html
```
It's actually a redirect to this page

File: app/config/security_test.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+security:
+  # http://symfony.com/doc/current/book/security.html#where-do-users-come-from-user-providers
```
Comment (matks) [inline app/config/security_test.yml]: Same

**Comment by mflasquin** [comment]
Test UI @Progi1984 https://github.com/mflasquin/ga.tests.ui.pr/actions/runs/4490764309/jobs/7898298304

**Review by jolelievre — APPROVED** [review]

**Comment by Progi1984** [comment]
@mflasquin Please restart failed tests.

**Comment by mflasquin** [comment]
> @mflasquin Please restart failed tests.

@Progi1984  Done https://github.com/mflasquin/ga.tests.ui.pr/actions/runs/4490764309

**Comment by Progi1984** [comment]
Yes, it's :green_circle:.

@PrestaShop/qa-functional : This PR fixes #31694. Are you agree if I put the "QA :heavy_check_mark:" label ?

**Review by djoelleuch — APPROVED** [review]
Hi @Progi1984 , 
it is QA approved , you can put the  QA :heavy_check_mark:

**Review by matks — APPROVED** [review]

**Comment by mflasquin** [comment]
thanks all :smile:

**Pull request merged**